File: leetcode-python-go/1945. 字符串转化后的各位数字之和/main.py
s = "leetcode"
k = 2

def getLuckys(s: str, k: int) -> int:
    digits = "".join(str(ord(ch) - ord("a") + 1) for ch in s)

    for i in range(k):
        if len(digits) == 1:
            break
        total = sum(int(ch) for ch in digits)
        digits = str(total)
    
    return int(digits)

print(getLuckys(s,k))

File: leetcode-python-go/1945. 字符串转化后的各位数字之和/main.go
package main

import (
	"fmt"
	"strconv"
)

func getLucky(s string, k int) int {
	t := []byte{}
	for _, c := range s {
		t = append(t, strconv.Itoa(int(c-'a'+1))...)
	}
	digits := string(t)
	for i := 1; i <= k && len(digits) > 1; i++ {
		sum := 0
		for _, c := range digits {
			sum += int(c - '0')
		}
		digits = strconv.Itoa(sum)
	}
	ans, _ := strconv.Atoi(digits)
	return ans
}

func main() {
	s := "leetcode"
	k := 2
	results := getLucky(s, k)
	fmt.Println(results)
}
